be/src/exec/vectorized/hdfs_scanner.cpp

```diff
@@ -174,6 +174,37 @@ void HdfsScanner::update_counter() {
     }
 }
 
+void HdfsScanner::update_counter() {
+#ifndef BE_TEST
+
+    update_hdfs_counter();
+
+    HdfsScanProfile* profile = _scanner_params.profile;
+    COUNTER_UPDATE(profile->scan_timer, _stats.scan_ns);
+    COUNTER_UPDATE(profile->reader_init_timer, _stats.reader_init_ns);
+
+    COUNTER_UPDATE(profile->rows_read_counter, _stats.raw_rows_read);
+    COUNTER_UPDATE(profile->bytes_read_counter, _stats.bytes_read);
+    COUNTER_UPDATE(profile->expr_filter_timer, _stats.expr_filter_ns);
+    COUNTER_UPDATE(profile->io_timer, _stats.io_ns);
+    COUNTER_UPDATE(profile->io_counter, _stats.io_count);
+    COUNTER_UPDATE(profile->column_read_timer, _stats.column_read_ns);
+    COUNTER_UPDATE(profile->column_convert_timer, _stats.column_convert_ns);
+
+    HdfsParquetProfile* parquet_profile = profile->parquet_profile;
```

Comment:
Can `HdfsParquetProfile` inherit `HdfsScanProfile`?
Common profiles can be updated in `hdfs_scanner` and profiles related to parquet or orc can be updated in the subclass.

Reply:
Yes. It's quite hard. But each scanner can update it's own profile.